### PR TITLE
Use standard CSS rules to resize emoticons

### PR DESCRIPTION
**Pull request by lorumic**
#### Summary

Fixes a CSS issue in an upcoming Firefox version. Currently, a `-moz` prefixed `transform: scale` is applied together with non-standard `zoom` rule, causing them to be both applied at the same time, which makes emoticon sprites too small on Firefox 133.

Instead of the vendor-prefixed standard rule combined with a non-standard one, it is better to use the standard counterpart of `zoom`, which is `transform: scale`.

Original context of the issue can be found [here](https://github.com/mattermost/mattermost-webapp/pull/11936). However, as of the latest version of Chrome and Firefox, there does not seem to be a noticeable difference between `transform: scale` and `zoom` in terms of "blurriness", so non-standard `zoom` rule can probably be replaced with `transform: scale`.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/29114
Jira https://mattermost.atlassian.net/browse/MM-61527

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by lorumic** [comment]
> @hanzei I'm removing myself from reviewing this. Not sure if I see a difference in the changes.

@rahimrahman You can find the reproduction steps in the [linked issue](https://github.com/mattermost/mattermost/issues/29114). Then the difference should become apparent.

**Comment by lorumic** [comment]
> I understand, it looks good to me based on screenshots. But I have no real understanding of why `zoom` was used in the first place here. I'm deferring to my colleague with a bit more context especially from the comment mentioning [MM-34178](https://mattermost.atlassian.net/browse/MM-34178).

I see, thanks for the reference. I wasn't aware that the original issue was on macOS. Unfortunately I'm on Linux, so I cannot check if this change is causing a regression in terms of "blurriness" on macOS.

**Comment by wezm** [comment]
Mattermost folks can we get this merged? The tiny emoji are really hard to see and make the picker difficult to use.

tiny-emoji

**Comment by rahimrahman** [comment]
@Willyfrog @hmhealey can you all help review this or assign somebody?  I was the only reviewers assigned by @mattermost/core-reviewers and I'm 0/5 on the use of zoom vs scale which may cause regression (see this [jira ticket](https://mattermost.atlassian.net/browse/MM-34178))

File: webapp/channels/build/emoji/make_emojis.mjs
Comment (rahimrahman) [inline webapp/channels/build/emoji/make_emojis.mjs:446]: @lorumic Maybe we can do both here?  Do you mind testing on both?

```suggestion
    zoom: 0.5;  /* fixes blurry images on Chrome? */
    transform: scale(0.5); /* fixes small images on Firefox? */
```

Comment (lorumic) [reply webapp/channels/build/emoji/make_emojis.mjs:446]: The problem is that if you do both, both get applied on FF 133, resulting in the issue that was reported: 0.5 from `zoom` and 0.5 from `transform: scale` results in a final 0.25 size.
I think the takeaway here is that the usage of `zoom` should not be mixed with that of `transform: scale`. You could use `zoom` only, but that would probably affect your browser compatibility: [scale](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/scale#browser_compatibility) vs [zoom](https://developer.mozilla.org/en-US/docs/Web/CSS/zoom#browser_compatibility).

Comment (rahimrahman) [reply webapp/channels/build/emoji/make_emojis.mjs:446]: Ah thank you.  I acknowledge that my knowledge of css is subpar, thus I'm seeking others to help.  I'm just throwing a few ideas.

Perhaps we can use `@supports`?  Again, this may not work as I don't spend much time on web css (I'm more of a mobile engineer here at MM).

```
.zoomable {
  zoom: 0.5; /* Default for browsers supporting zoom */
}

@supports (transform: scale(1)) {
  .zoomable {
    zoom: unset; /* Reset zoom for modern browsers */
    transform: scale(0.5); /* Use scale instead */
    transform-origin: 0 0; /* Anchor scaling */
  }
}
```

Comment (lorumic) [reply webapp/channels/build/emoji/make_emojis.mjs:443]: This is a good idea, but it should be the other way around, as `zoom` is the less-supported property. :)
I have just updated the code to use `zoom` whenever supported (which means "always" in Chrome, where the blurriness problem originally occurred, because Chrome supports it since version 1), and fall back to `transform: scale` when unsupported (e.g. FF versions before 126).
Can you maybe have another look @rahimrahman?

Comment (rahimrahman) [reply webapp/channels/build/emoji/make_emojis.mjs:443]: Smart, totally see your reasoning of reversing.  Of course, my example would have cause blurriness on Chrome and totally not fixing the problem.  😏 

Do you have a screenshot of Chrome which shows no blurriness and another on Firefox that shows the icon is the correct size?  Visual representation helps a lot, especially for other reviewers.

Thanks for working to solve this even at small **scale** (pun intended!).  I appreciate helping me understand the diff between zoom and scale.  Something I have not encountered before (again as a mobile dev, I don't spend a lot of time here).

Comment (rahimrahman) [reply webapp/channels/build/emoji/make_emojis.mjs:443]: Just re-read that you're on Linux.  Let me check on mine, or we can get one of our QAs to verify.

**Comment by rahimrahman** [comment]
@lindy65 I've invited you to this PR to help me verify that the changes by @lorumic help solve the small icon issue on Firefox v133 (latest, as of today), and blurry icon on Chrome (any version).

Thanks.

If I can get to this first, I would update you.

**Comment by lindy65** [comment]
Hey @lindalumitchell - I've added you for review on this PR to check the fix on macOS Chrome, thanks!

**Comment by lindy65** [comment]
Hi @lorumic - there are some failing checks - can you take a look please? I'm not able to set up a test server until all checks pass. Thanks!
cc @rahimrahman

**Comment by lorumic** [comment]
> Hi @lorumic - there are some failing checks - can you take a look please? I'm not able to set up a test server until all checks pass. Thanks! cc @rahimrahman

Hi @lindy65 - sorry for my carelessness. :) The linting issue should be fixed now.

**Review by lindy65 — APPROVED** [review]
Thanks @lorumic - tested as per repro steps in the ticket and the emoji issue in Windows Chrome and Windows Firefox 133 is fixed / emoji are the correct size.

I do not have Linux so was wondering whether you're able to verify the fix with a screenshot on Linux?

I've asked my colleague, @lindalumitchell to test on mac

cc @rahimrahman

**Comment by lorumic** [comment]
> I do not have Linux so was wondering whether you're able to verify the fix with a screenshot on Linux?

Sure, here's how it looks like on Linux:

- Chrome:

image

- Firefox 133:

image

**Comment by lindy65** [comment]
Fantastic @lorumic - thanks so much for providing the screenshots 👍

**Review by rahimrahman — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]
LGTM, I did a quick test as well. Thank you for taking the time to research and fix this!

**Comment by rahimrahman** [comment]
@lorumic thanks for the discussion and your contribution.  It was nice working with you.  I learned a ton.  If you want to explore the mobile side of things, hit me up in Community.

@BenCookie95 I will merge, but I don't know if there's anything special I have to do here to get to v10.4 (which I believe 12/2 was the code freeze).

@lindy65 is there anything we need to do to kill the test server?

**Comment by lindy65** [comment]
>@lindy65 is there anything we need to do to kill the test server?

Hey @rahimrahman - removing the test server label will destroy the server but it's also destroyed on merge

**Comment by felixb-denic** [comment]
Hey,
I don't know anything about the internal release mechanism of Mattermost, but is there a reason why this change hasn't made it's way into a release? At least on the ESR branch we already had two releases since merging, and neither 9.11.6 nor 9.11.7 contained this fix.

Obviously not a massive bug, I'm just curious why a already merged PR from over a month ago takes so long to reach the customer.

**Comment by lindy65** [comment]
> Hey, I don't know anything about the internal release mechanism of Mattermost, but is there a reason why this change hasn't made it's way into a release? At least on the ESR branch we already had two releases since merging, and neither 9.11.6 nor 9.11.7 contained this fix.
> 
> Obviously not a massive bug, I'm just curious why a already merged PR from over a month ago takes so long to reach the customer.

Hey @felixb-denic - this fix is scheduled to be released today in v10.4

**Comment by felixb-denic** [comment]
> Hey @felixb-denic - this fix is scheduled to be released today in v10.4

Good to know, thanks.
But what about ESR? We still have nearly have of the lifetime of 9.11 ahead of us. Does this fix will get backported?

**Comment by lindy65** [comment]
> > Hey @felixb-denic - this fix is scheduled to be released today in v10.4
> 
> Good to know, thanks. But what about ESR? We still have nearly have of the lifetime of 9.11 ahead of us. Does this fix will get backported?

It's not currently scheduled to be back-ported but our release manager, @amyblais will be able to help with this

**Comment by ThiefMaster** [comment]
Backporting would be nice, especially considering how trivial the fix it (for self-builds it's a simple cherry pick to backport it yourself, but I guess the typical Enterprise user of ESR can't do that)

**Comment by amyblais** [comment]
Sounds good, we can take a look at backporting to v9.11.

**Comment by felixb-denic** [comment]
Thanks @amyblais  :)

Can confirm it's part of 9.11.8

**Pull request merged**
